pkg/nfd-master/nfd-master.go

```diff
@@ -1447,19 +1419,27 @@ func (m *nfdMaster) filterFeatureAnnotations(annotations map[string]string) map[
 			continue
 		}
 		// Check annotation namespace, filter out if ns is not whitelisted
-		if ns != nfdv1alpha1.FeatureAnnotationNs && !strings.HasSuffix(ns, nfdv1alpha1.FeatureAnnotationSubNsSuffix) {
-			// If the namespace is denied the annotation will be ignored
-			if ns == "" {
-				klog.ErrorS(fmt.Errorf("labels without namespace (prefix/) not allowed"), fmt.Sprintf("Ignoring annotation %s", annotation))
-				continue
-			}
-			if ns == "kubernetes.io" || strings.HasSuffix(ns, ".kubernetes.io") || ns == nfdv1alpha1.AnnotationNs {
-				klog.ErrorS(fmt.Errorf("namespace %q is not allowed", ns), fmt.Sprintf("Ignoring annotation %s", annotation))
-				continue
-			}
+		err := validate.Annotation(annotation, value)
+		if err != nil {
+			klog.ErrorS(err, "ignoring annotation", "annotationKey", annotation, "annotationValue", value)
+			continue
 		}
 
 		outAnnotations[annotation] = value
 	}
 	return outAnnotations
 }
+
+func isNamespaceDenied(labelNs string, wildcardDeniedNs map[string]struct{}, normalDeniedNs map[string]struct{}) bool {
```

Comment:
nit: why was this func moved? Harder to review (and tell if smth changed) if things are moved around. If we want to do this put it into a separate commit for example to isolate the change